apps/jobs/utils.py

```diff
@@ -334,6 +334,9 @@ def calculate_distinct_sorted_leaderboard_data(
 
     all_banned_participant_team = []
     for leaderboard_item in leaderboard_data:
+        if challenge_phase_split.show_execution_time:
```

Comment:
Hey @burnerlee, Can we please add this `if` a little above i.e. before this line `leaderboard_data = leaderboard_data.annotate` and create the annotated list based on this flag. Here you are again querying the database `n` times for `n` entries in the leaderboard. This will make things slow when there is a large number of entries on the leaderboard.